pr_agent/tools/pr_reviewer.py

```diff
@@ -103,4 +108,37 @@ def _prepare_pr_review(self) -> str:
 
         if settings.config.verbosity_level >= 2:
             logging.info(f"Markdown response:\n{markdown_text}")
-        return markdown_text
\ No newline at end of file
+        return markdown_text
+
+    def _publish_inline_code_comments(self):
+        if settings.config.git_provider != 'github': # inline comments are currently only supported for github
+            return
+
+        review = self.prediction.strip()
+        try:
+            data = json.loads(review)
+        except json.decoder.JSONDecodeError:
+            data = try_fix_json(review)
+
+        pr = self.git_provider.pr
+        last_commit_id = list(pr.get_commits())[-1]
+        files = list(self.git_provider.get_diff_files())
+
+        for d in data['PR Feedback']['Code suggestions']:
+            relevant_file = d['relevant file'].strip()
+            relevant_line_in_file = d['relevant line in file'].strip()
+            content = d['suggestion content']
+            position = -1
+            for file in files:
+                if file.filename.strip() == relevant_file:
+                    patch = file.patch
+                    patch_lines = patch.splitlines()
+                    for i, line in enumerate(patch_lines):
+                        if relevant_line_in_file in line:
+                            position = i
+            if position == -1:
+                logging.info(f"Could not find position for {relevant_file} {relevant_line_in_file}")
+            else:
+                body = content
+                path = relevant_file.strip()
+                pr.create_review_comment(body=body, commit_id=last_commit_id, path=path, position=position)
```

Comment:
**Suggestion:** Consider handling the case where `position` is not found in the patch lines. This will prevent creating a review comment at an invalid position.
```suggestion
            if position == -1:
                logging.info(f'Could not find position for {relevant_file} {relevant_line_in_file}')
                return
            body = content
            path = relevant_file.strip()
            pr.create_review_comment(body=body, commit_id=last_commit_id, path=path, position=position)
```